generator/src/main/java/com/scottlogic/deg/generator/decisiontree/ConstraintNodeBuilder.java

```diff
@@ -0,0 +1,107 @@
+package com.scottlogic.deg.generator.decisiontree;
+
+import com.scottlogic.deg.common.profile.constraints.atomic.AtomicConstraint;
+import com.scottlogic.deg.common.util.FlatMappingSpliterator;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static java.util.stream.Stream.concat;
+
+public class ConstraintNodeBuilder {
+    private final Collection<AtomicConstraint> atomicConstraints;
+    private final Collection<DecisionNode> decisions;
+    private final Set<NodeMarking> nodeMarkings;
+
+    public ConstraintNodeBuilder(Collection<AtomicConstraint> atomicConstraints, Collection<DecisionNode> decisions, Set<NodeMarking> nodeMarkings) {
```

Comment:
It might be nicer to make this constructor private, and construct every constraint node up from the base one. This might be a good shortcut method to have though, so it's up to you.

Reply:
Hmm... as you pointed out, this is quite a useful shortcut used by constraintNode to get a corresponding builder. I think I will leave it public for now.